### PR TITLE
Recommendation skeleton

### PR DESCRIPTION
**Pull request by Boris713**
Changes:
1. Added user info to database from firebase sign up
2. Added recommendation system 
3. Added API end points for recommendation system

All changes internal no images

### CODE REVIEW

File: client/src/pages/Authentication/Authentication.jsx [hunk 1]
```diff
@@ -27,7 +27,6 @@ const Authentication = () => {
       try {
         await doSignInWithEmailAndPassword(logInEmail, logInPassword);
         navigate("/home");
-        console.log("signed in");
```
Comment (akhilmandadi) [inline client/src/pages/Authentication/Authentication.jsx:30]: remove consoles while pushing

File: client/src/pages/Authentication/Authentication.jsx [hunk 2]
```diff
@@ -39,12 +38,34 @@ const Authentication = () => {
...
-      ).catch((err) => {
+      try {
+        const credentials = await doCreateUserWithEmailAndPassword(
+          registerEmail,
+          registerPassword
+        );
+        const id = credentials.user.uid;
+        const email = credentials.user.email;
+
+        fetch("http://localhost:3000/itinerary/create-user", {
```
Comment (akhilmandadi) [inline client/src/pages/Authentication/Authentication.jsx]: try to have your base url http://localhost:3000 stored in your env

File: server/python/recommend.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import pandas as pd
+import numpy as np
+from scipy.sparse.linalg import svds
+
+df = pd.read_csv("user_activity_data.csv")
```
Comment (akhilmandadi) [inline server/python/recommend.py:5]: why is the data read from csv?

File: server/routes/routes.js [hunk 4]
```diff
@@ -3,6 +3,10 @@ const router = express.Router();
 const { PrismaClient } = require("@prisma/client");
 const prisma = new PrismaClient();
 const axios = require("axios").default;
+const { exec } = require("child_process");
+const { bodyParser } = require("body-parser");
+const { error } = require("console");
+const { stdout, stderr } = require("process");
 
 router.get("/activities", async (req, res) => {
```
Comment (akhilmandadi) [inline server/routes/routes.js:11]: move routes into router files as you did for codepath project

File: server/routes/routes.js [hunk 5]
```diff
@@ -24,4 +28,64 @@ router.get("/activities", async (req, res) => {
     res.status(500).send("Failed to fetch data");
   }
 });
+
+router.post("/create-user", async (req, res) => {
```
Comment (akhilmandadi) [inline server/routes/routes.js]: /user on POST will denote that is a create user operation. please rename it

File: server/routes/routes.js [hunk 6]
```diff
@@ -24,4 +28,64 @@ router.get("/activities", async (req, res) => {
...
+      data: {
+        id: id,
+        email: email,
+        name: name || "User",
+      },
+    });
+    res.status(201).json(user);
+  } catch (error) {
+    if (error.code === "P2002") {
+      res.status(409).send("A user with this email already exists.");
```
Comment (akhilmandadi) [inline server/routes/routes.js]: try to send json for error responses as well. Would be easy to parse in frontend

**Review by akhilmandadi — CHANGES_REQUESTED** [review]

File: client/src/pages/Authentication/Authentication.jsx (hunk 1 above)
Comment (Boris713) [reply client/src/pages/Authentication/Authentication.jsx:30]: had removed after self review. Ill review before making pr in future!

File: server/routes/routes.js (hunk 5 above)
Comment (Boris713) [reply server/routes/routes.js]: do you mean rename to something like /register?

File: server/python/recommend.py (hunk 3 above)
Comment (Boris713) [reply server/python/recommend.py:5]: csv contains needed info

**Pull request merged**